### ISSUE TITLE
`ApplicationServices` framework is not available on `iOS`?

### ISSUE THREAD

**Issue by st235** [libsdl-org/SDL_image#470]
It appears that the `ApplicationServices` framework is linked across the entire `Apple` ecosystem in the [CMakeLists.txt](https://github.com/libsdl-org/SDL_image/blob/main/CMakeLists.txt#L371). When building for `iOS` both `IOS` and `APPLE` flags are defined, therefore `ApplicationServices` is linked to the target. 

However, I believe it is not available on `iOS`, as generating an `Xcode` project with `SDL3_image` linked for iOS results in a build failure.

![Screenshot 2024-09-29 at 17 24 19](https://github.com/user-attachments/assets/29c9149d-9948-4cd2-9aaf-1fc953a071d6)

I am wondering whether I am doing anything wrong or this is really a bug in the build script?

This is my `CMakeLists.txt` build script if of any use (some details are lifted for the sake of simplicity):

```cmake
FetchContent_Declare(
    SDL_IMAGE
    GIT_REPOSITORY https://github.com/libsdl-org/SDL_image.git
    GIT_TAG main
    GIT_SHALLOW TRUE
    GIT_PROGRESS TRUE
)

FetchContent_MakeAvailable(SDL_IMAGE)

add_library(core INTERFACE)
target_sources(core INTERFACE
  src/app.h
  src/app.cpp
)
target_include_directories(core INTERFACE ${CMAKE_CURRENT_LIST_DIR}/include)
target_link_libraries(core INTERFACE SDL3::SDL3 SDL3_image::SDL3_image)
```

And this is the build command I use

```bash
cmake -B out/ios -GXcode -DCMAKE_SYSTEM_NAME=iOS -DCMAKE_OSX_ARCHITECTURES="arm64;i386;x86_64"
```

P.S.: the declaration of `SDLIMAGE_BACKEND_IMAGEIO` slightly hints that this is only related to `Mac OS X`.

```cmake
cmake_dependent_option(SDLIMAGE_BACKEND_IMAGEIO "Use native Mac OS X frameworks for loading images" ON APPLE OFF)
```

P.P.S.: [this stack overflow thread](https://stackoverflow.com/questions/9393295/applicationservices-framework-not-found-error-in-ios) also suggests that `ApplicationServices` are not available in `iOS`.

P.P.P.S.: there is also [an old unresolved SDL community thread](https://discourse.libsdl.org/t/sdl-ios-demo-application-build-failure-applicationservices-h-file-not-found/31947) with the similar issue

**Comment by madebr** [comment]
https://github.com/libsdl-org/SDL_image/blob/9d303c3906c31ac8cc6532268179df2bce4babca/CMakeLists.txt#L352

How does one use imageio on iphoneos? Is it available by default?

**Comment by slouken** [comment]
It's used on macOS for CGImage functionality.

**Comment by slouken** [comment]
```c
#if (TARGET_OS_IPHONE == 1) || (TARGET_IPHONE_SIMULATOR == 1)
#import <MobileCoreServices/MobileCoreServices.h> // for UTCoreTypes.h
#import <ImageIO/ImageIO.h>
#import <UIKit/UIImage.h>
#else
// For ImageIO framework and also LaunchServices framework (for UTIs)
#include <ApplicationServices/ApplicationServices.h>
#endif
```

**Comment by madebr** [comment]
Fixed in ee91574696dadc386e708ed4f0c5bbd89798426e, and backported to SDL2 + 2.8.x

**Comment by st235** [comment]
@madebr @slouken Hey folks 👋

Thanks for looking into the issue!
It seems the original problem has been resolved, but there’s still no alternative `iOS` framework specified for building on `iOS`.

I've opened a PR to address the `iOS` build issues as well: https://github.com/libsdl-org/SDL_image/pull/553
